ui/components/app/assets/hooks/useTokenDisplayInfo.tsx

```diff
@@ -0,0 +1,111 @@
+import { getIntlLocale } from '../../../../ducks/locale/locale';
+import { formatAmount } from '../../../../pages/confirmations/components/simulation-details/formatAmount';
+import { getCurrentCurrency } from '../../../../ducks/metamask/metamask';
+import { useShouldShowFiat } from '.';
+
+type UseTokenDisplayInfoProps = {
+  token: TokenWithFiatAmount;
+};
+
+type TokenDisplayInfo = {
+  title: string;
+  tokenImage: string;
+  primary: string;
+  secondary: string | undefined;
+  isStakeable: boolean | undefined;
+  tokenChainImage: string;
+};
+
+export const useTokenDisplayInfo = ({
+  token,
+}: UseTokenDisplayInfoProps): TokenDisplayInfo => {
+  const isEvm = useSelector(getMultichainIsEvm);
+  const tokenList = useSelector(getTokenList);
+  const erc20TokensByChain = useSelector(selectERC20TokensByChain);
+  const shouldShowFiat = useShouldShowFiat(); // TODO: break out currency formatter into a useFormattedFiatHook (chain agnostic)
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const locale = useSelector(getIntlLocale);
+  const tokenChainImage = getImageForChainId(token?.chainId);
+
+  const secondaryThreshold = 0.01;
+
+  // Format for fiat balance with currency style
+  const secondary =
+    shouldShowFiat && token.tokenFiatAmount
+      ? formatWithThreshold(
+          Number(token.tokenFiatAmount),
+          secondaryThreshold,
+          locale,
+          {
+            style: 'currency',
+            currency: currentCurrency.toUpperCase(),
+          },
+        )
+      : undefined;
+
+  const primary = formatAmount(
+    locale,
+    new BigNumber(Number(token.string) || '0', 10),
+  );
+
+  const isEvmMainnet =
+    token.chainId && isEvm ? isChainIdMainnet(token.chainId) : false;
+
+  const isStakeable = isEvmMainnet && isEvm && token.isNative;
+
+  if (isEvm) {
+    const tokenData = Object.values(tokenList).find(
+      (tokenToFind) =>
+        isEqualCaseInsensitive(tokenToFind.symbol, token.symbol) &&
+        isEqualCaseInsensitive(tokenToFind.address, token.address),
+    );
+
+    const title =
+      tokenData?.name ||
+      (token.chainId === '0x1' && token.symbol === 'ETH'
+        ? 'Ethereum'
+        : token.chainId &&
+          erc20TokensByChain?.[token.chainId]?.data?.[
+            token.address.toLowerCase()
+          ]?.name) ||
+      token.symbol;
+
+    const tokenImage =
+      tokenData?.iconUrl ||
+      (token.chainId &&
+        erc20TokensByChain?.[token.chainId]?.data?.[token.address.toLowerCase()]
+          ?.iconUrl) ||
+      token.image;
+
+    return {
+      title,
+      tokenImage,
+      primary,
+      secondary,
+      isStakeable,
+      tokenChainImage,
+    };
+  }
+  // TODO non-evm assets. this is only the native token
+  return {
+    title: token.symbol,
+    tokenImage: token.image,
+    primary: '',
+    secondary: token.secondary,
+    isStakeable: false,
+    tokenChainImage: token.image,
+  };
```

Comment:
@salimtb @sahar-fehri This is where we will map non-evm/non-native tokens to their token metadata. It should include these properties to get passed into props for the remainder of the component tree.

This may not be our final solution, but hope this helps to make future migrations easier.